README.md

```diff
@@ -42,3 +42,10 @@ For example, if the instance connection name is `foo:bar:baz`, the JDBC URL
 would be 
 `jdbc:mysql://google/mydb?cloudSqlInstance=foo:bar:baz&socketFactory=com.google.cloud.sql.mysql.SocketFactory`
 
+## Credentials
```

Comment:
Do we want to add docs about Kubernetes as well?

Specifically, k8s doesn't recommend using default credentials even when running on GCE. Instead, they suggest downloading a service account and setting the environment variable for controlling credentials.